p2p/net/conn/listen.go

```diff
@@ -157,12 +135,62 @@ func (l *listener) Loggable() map[string]interface{} {
 	}
 }
 
+func (l *listener) handleIncoming() {
+	var wg sync.WaitGroup
+	defer func() {
+		wg.Wait()
+		close(l.incoming)
+	}()
+
```

Comment:
... hmmm well, it's not concurrency safe. https://github.com/jbenet/go-temp-err-catcher/blob/master/temp_err_catcher.go#L82